shapeshift/web
# feat: cosmos staking bonding states

## Description

I modified `<StackingOpprtunities>` component in order to handle validator with stacked tokens. I made the choice of re-using <ReactTable> for this, I prefered not to re-invent the wheel too much ... It is very convenient for the stacking case, a little bit less in the non-stacking case.

## Notice




## Pull Request Type


## Issue (if applicable)

Issue is https://github.com/shapeshift/web/issues/1267, wiring up with real data is missing

## Risk

The wiring with real data is missing but will be done in a separate PR once available in lib / web.
Also, <ReactTable> component has been slightly modified. This should not impact other areas of the code that are making use of it (I verified earn opportunities page and it is working fine).

## Testing

- Go to OSMO asset page
- Check staking opportunities is conform to the expectation
- Changes data in code to have only validator without any stacking
- Verify staking opportunities is still conform to the expectation

## Screenshots (if applicable)

![stacking](https://user-images.githubusercontent.com/5720927/160290596-85dc715b-66dc-497d-90d3-27484dd19135.png)

![nostacking](https://user-images.githubusercontent.com/5720927/160290604-e8e1c905-9d0c-49b9-a9ff-c932e225c9b6.png)
